In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
from math import sqrt

data = {
    'student_id': [1, 2, 3, 4, 5],
    'attendance_rate': [90, 80, 85, 70, 95],
    'average_grade_previous': [75, 80, 78, 70, 88],
    'socio_emotional_score': [85, 75, 80, 65, 90],
    'current_grade': [78, 82, 79, 72, 90]
}

df = pd.DataFrame(data)

# Defining features and target variable
X = df[['attendance_rate', 'average_grade_previous', 'socio_emotional_score']]
y = df['current_grade']

# Splitting the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Initializing and training the linear regression model
model = LinearRegression()
model.fit(X_train, y_train)

# Predicting on the test set
y_pred = model.predict(X_test)

# Evaluating the model
rmse = sqrt(mean_squared_error(y_test, y_pred))
print(f'Root Mean Squared Error: {rmse}')

Root Mean Squared Error: 0.6542372881356044


This indicates that, on average, the model's predictions deviate from the actual current grades by roughly 0.6542 points.